### ISSUE TITLE
How can we get both server response and upload progress using `UploadTask`?

### ISSUE THREAD

**Issue by minhhoangtn** [781flyingdutchman/background_downloader#100]
Currently, we can only get the server response using `await FileDownloader().request`.

But let's assume we want to upload the file to the server, track the progress and get the file path on the server from the response from the request, can you please show me how to achieve both the upload progress and server response?

**Comment by 781flyingdutchman** [comment]
You're right that that is currently not possible. Can you elaborate on what you mean with the server response? You mention file path but perhaps you're referring to the `body` of the server response (which is what's accessible using `Requwat`?

**Comment by minhhoangtn** [comment]
Yes, normally after uploading files to the remote server, we will receive a URL to access the file in the `body` of the server response.

For example, I have the API endpoint to upload files like this, and I will need to access the `body` of the server response for the next API (which is only accessible using `await FileDownloader().request()`
<img width="981" alt="Screenshot 2023-07-24 at 13 08 13" src="https://github.com/781flyingdutchman/background_downloader/assets/89381951/094ac87f-9d2a-4be1-8fa2-5ee7b4659f5b">

**Comment by 781flyingdutchman** [comment]
Cool, that makes sense. I think we can make `responseBody` available  as a field in `TaskStatusUpdate` and it would only be relevant for status `TaskStatus.complete`. Would that work for your use case?

**Comment by minhhoangtn** [comment]
Yeah, I think it will work perfectly for my case. Can you tell me when we have a new update for this enhancement? It would be a big help to me. Thank you

**Comment by 781flyingdutchman** [comment]
Gimme a minute 😁

**Comment by 781flyingdutchman** [comment]
Hi @minhhoangtn I've implemented this on the `server_response` branch, so change your pubspec.yaml to:
```
background_downloader:
    git:
      url: https://github.com/781flyingdutchman/background_downloader.git
      ref: server_response
```

When you get a `TaskStatusUpdate` (eg via the Listener or in response to a `FileDownloader().upload` call) that update now includes a field `responseBody` that, if not null, contains the server response.  The responseBody will also contain a server response when certain errors are given, e.g. for a 404 (not found) status code, also for download tasks.  I think that also fixes issue #80 

Can you please check this out and test it, and let me know if it works as you would expect?  Once it's ok, I'll publish. Thanks!

**Comment by minhhoangtn** [comment]
Hi @781flyingdutchman, thanks so much for your hard work. I'll start to test it and back to you the result if it works as expected asap.

**Comment by minhhoangtn** [comment]
@781flyingdutchman Hi, after testing it, I think it works as expected. I'm not 100% sure but I think there is a small issue with the `onProgress` callback in `FileDownloader().upload()` on the Android platform.

 I print the progress value when I upload a file and it seems the value is reduced from **0** to **-1** as the upload progresses. While on iOS, the value is increased from **0** to **1**.

This is my log on Android:
```
I/BackgroundDownloader( 5863): Enqueuing task with id 1645254828
D/EGL_emulation( 5863): app_time_stats: avg=631.17ms min=5.50ms max=1851.31ms count=3
I/TaskWorker( 5863): Starting task with taskId 1645254828
I/flutter ( 5863): progress: 0.0
W/TaskWorker( 5863): Error for taskId 1645254828 and /data/user/0/co.example.dev/cache/NewWords.txt: Cleartext HTTP traffic to 206.189.154.242 not permitted
I/flutter ( 5863): progress: -1.0
W/System  ( 5863): A resource failed to call close.
I/WM-WorkerWrapper( 5863): Worker result SUCCESS for Work [ id=d2cce220-a901-4afd-9c22-35976b26ea40, tags={ com.bbflight.background_downloader.TaskWorker, BackgroundDownloader, taskId=1645254828, group=await } ]
```

**Comment by 781flyingdutchman** [comment]
Thanks for testing! The Android result is actually correct, in that - according to the logs - your task fails, because it uses http instead of https (that's the Cleartext traffic reference in the error line in your log). The progress value for a "failed" status is -1. If you use https things should work the same on Android, but let me know if you're running into something.

**Comment by 781flyingdutchman** [comment]
Included in V7.8.0
